zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterFactory.java

```diff
@@ -856,6 +856,23 @@ public void setPropertyAndRestart(String id, InterpreterOption option, Propertie
     }
   }
 
+  private boolean interpreterIsShared(InterpreterSetting setting) {
+    if (setting.getOption().isPerNoteProcess() == false &&
+      setting.getOption().isPerNoteSession() == false){
+      return true;
+    }
+    return false;
```

Comment:
Thank you for the review.

1. I totally agree with your opinion! but lots of zeppelin codes have same problems. 
Do you think it should be refactored the other codes too?

2. The interpreterIsShared() method returns `true` only when two variables are same `false`, Is it same with `return interpreterSetting.isPerNoteProcess() || interpreterSetting.isPerNoteSession()` ?